pyparsing/results.py

```diff
@@ -66,41 +66,41 @@ def test(s, fn=repr):
         - year: 1999
     """
 
+    __slots__ = [
+        "_result__name", "_result__parent", "_result__all_names",
+        "_result__modal", "_result__list", "_result__dict", "__weakref__"
+    ]
+
     def __new__(cls, toklist=None, name=None, asList=True, modal=True):
         if isinstance(toklist, ParseResults):
             return toklist
-        retobj = object.__new__(cls)
-        retobj.__doinit = True
-        return retobj
+        self = object.__new__(cls)
+        self._result__name = None
+        self._result__parent = None
+        self._result__all_names = set()
+        self._result__modal = modal
+        if toklist is None:
+            toklist = []
+        if isinstance(toklist, list):
+            self._result__list = toklist[:]
+        elif isinstance(toklist, _generator_type):
+            self._result__list = list(toklist)
+        else:
+            self._result__list = [toklist]
+        self._result__dict = dict()
+        return self
 
     # Performance tuning: we construct a *lot* of these, so keep this
     # constructor as small and fast as possible
     def __init__(
         self, toklist=None, name=None, asList=True, modal=True, isinstance=isinstance
     ):
-        if self.__doinit:
-            self.__doinit = False
-            self.__name = None
-            self.__parent = None
-            self.__accumNames = {}
-            self.__asList = asList
-            self.__modal = modal
-            if toklist is None:
-                toklist = []
-            if isinstance(toklist, list):
-                self.__toklist = toklist[:]
-            elif isinstance(toklist, _generator_type):
-                self.__toklist = list(toklist)
-            else:
-                self.__toklist = [toklist]
-            self.__tokdict = dict()
-
         if name is not None and name:
             if not modal:
-                self.__accumNames[name] = 0
+                self._result__all_names = {name}
```

Comment:
Since `0` is the only value, we use a set

Reply:
Ha! That is how old this code is, pre-dates `set`. Then was kept to maintain compat with Python version before `set` (2.2? 2.3?). Nice catch!